### ISSUE TITLE
Differing behaviour with extra-index-urls between pyproject.toml and pixi.toml

### ISSUE THREAD

**Issue by ericmjl** [prefix-dev/pixi#2503]
### Checks

- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

`pyproject.toml` file looks like this:

```toml
[project]
name = "stuffy"
version = "0.0.1"
requires-python = ">=3.10"
dependencies = ["my-custom-package"]

[tool.pixi.project]
channels = ["conda-forge"]
platforms = ["linux-64"]

[tool.pixi.pypi-dependencies]
stuffy = { path = ".", editable = true }

# NOTE: Core dependencies (needed for running the program) go here.
[tool.pixi.dependencies]


[pypi-options]
extra-index-urls = ["https://url.to.my.custom.index/simple"]

```

Then I run the command to install the environment

```shell
❯ pixi install --manifest-path pyproject.toml -v
 INFO pixi_manifest::target: Adding pypi dependency: my-custom-package "*"
 INFO pixi_config: Loaded config from: /home/ema/.pixi/config.toml
 INFO pixi::environment: verifying prefix location is unchanged, with prefix file: /opt/mpc/stuffy/.pixi/envs/default/conda-meta/pixi_env_prefix
 INFO pixi::lock_file::outdated: environment 'default' is out of date because it does not exist in the lock-file.
 INFO pixi::lock_file::resolve::uv_resolution_context: uv keyring provider is disabled
 INFO resolve_conda{group=default platform=linux-64}: pixi::lock_file::update: fetched 2301 records in 969.243419ms
 INFO pixi::lock_file::update: resolved conda environment for environment 'default' 'linux-64' in 1s 381ms 229us 661ns
 INFO pixi::environment: No update needed for the prefix file.
 INFO pixi::environment: Verify history file exists: /opt/mpc/stuffy/.pixi/envs/default/conda-meta/history
 INFO pixi::lock_file::update: updated conda packages in the 'default' prefix in 6ms 47us 746ns
 INFO resolve_pypi{group=default platform=linux-64}: pixi::lock_file::resolve::pypi: there are no python packages installed by conda
  × failed to solve the pypi requirements of 'default' 'linux-64'
  ├─▶ failed to resolve pypi dependencies
  ╰─▶ Because my-custom-package was not found in the package registry and you require my-custom-package, we can conclude that your
      requirements are unsatisfiable.
```

Now I move the information into `pixi.toml`:

```toml
[project]
name = "stuffy"
version = "0.0.1"
channels = ["conda-forge"]
platforms = ["linux-64"]

[dependencies]
python = ">=3.10"

[pypi-dependencies]
my-custom-package = "*"

[pypi-options]
extra-index-urls = ["https://url.to.my.custom.index/simple"]
```

And we have no problems:

```bash
❯ pixi install --manifest-path pixi.toml
✔ The default environment has been installed.
```

### Issue description

I am unable to resolve the environment using the `pyproject.toml` file specification above, but am able to with `pixi.toml`. It appears there is a bug that is causing a difference between the two. Took me ~1 wk to narrow down the issue here, but I'm glad I was able to put in a minimal example (that at least works/doesn't work within my company's environment). (I did not use `-v` in the pixi command b/c that would have been overwhelming.)

My hunch here is that somehow, when using `pyproject.toml`, `uv` is somehow configured differently? When I try to use `uv` directly, it reads `pyproject.toml` and is capable of resolving versions that need to be installed.

I also tried configuring pixi globally:

```bash
❯ cat <<EOF > ~/.pixi/config.toml
[pypi-config]
extra-index-urls = ["https://url.to.my.custom.index/simple"]
EOF
```

### Expected behavior

Regardless of whether I am using `pyproject.toml` or `pixi.toml`, if their configurations are functionally equivalent, then I should be able to `pixi install` with no errors on any of them.

**Comment by tdejager** [comment]
Hi @ericmjl, sorry that this cost you this amount of work! Well it seems the `pyproject.toml` does not have the `[tool.pixi..]` prefix in front of the `pypi-options`. I suspect this would resolve things.

The global config only adds the options on init, it does not read those at runtime. This is to make the toml file as reproducible across machines as it can be.

**Comment by ericmjl** [comment]
@tdejager it looks like that did the trick! Thanks for the help. I ran into other errors w.r.t. authentication using ~/.netrc, but I will post a separate issue for this one.

**Comment by tdejager** [comment]
Hi @ericmjl! 

Again sorry for the struggles. We are planning to build something along the lines of what's requested here: https://github.com/prefix-dev/pixi/issues/2551

Should provide an easier option maybe :)

**Comment by ericmjl** [comment]
Oooh! I would love to upvote on this! But I have a question, since we are using uv underneath the hood, shouldn’t the use of environment variables be exposed already?

**Comment by tdejager** [comment]
@ericmjl you are correct! They are exposed and read by the uv library code, however they only work with, what I call, named indexes: https://docs.astral.sh/uv/configuration/indexes/ and https://docs.astral.sh/uv/configuration/indexes/#providing-credentials. I've checked the code and these are not applied to `extra-index-url` indexes, as of the last time I checked at least.

So I want to expose a similar feature in pixi, the only thing is we are a bit swamped currently with making `pixi build` a thing. But its pretty high on the priority list. Half of the PyPI issues might be authentication related and this should simplify things :D
